akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -72,6 +72,9 @@ sealed trait HttpEntity extends jm.HttpEntity {
       .via(new akka.http.impl.util.ToStrict(timeout, contentType))
       .runWith(Sink.head)
 
+  def discardEntityBytes(mat: Materializer): HttpMessage.DiscardedEntity =
```

Comment:
It's not really a preference thing, it's a "avoid unexpected weird situations thing".
Imagine you're in routing DSL and did:

```scala
extractEntity { e =>
  // I want to discard it...
  e.discard... // we're missing this
}
```

But maybe I did something with the raw request:

```scala
extractRequest { r =>
  // do stuff with `r`
  // discard entity
  r.discardEntity... // cool, we have this
```